gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobScheduler.java

```diff
@@ -357,19 +370,22 @@ public synchronized void handleNewJobConfigArrival(NewJobConfigArrivalEvent newJ
       }
     } catch (JobException je) {
       LOGGER.error("Failed to schedule or run job " + jobUri, je);
+      jobNameToNextSchedulableTime.put(jobUri, Instant.EPOCH);
     }
   }
 
   @Subscribe
   public synchronized void handleUpdateJobConfigArrival(UpdateJobConfigArrivalEvent updateJobArrival) {
     LOGGER.info("Received update for job configuration of job " + updateJobArrival.getJobName());
-    String jobName = updateJobArrival.getJobName();
+    String jobUri = updateJobArrival.getJobName();
```

Comment:
Let's not mix up the usage of job uri and job name. If you are gonna use job name (e.g. jobNameToNextSchedulableTime), then use the term job name everywhere. And if you are gonna use job uri, then change it for all of them